packages/core/src/views/notify/Notification.svelte

```diff
@@ -2,32 +2,149 @@
   import StatusIconBadge from './StatusIconBadge.svelte'
   import NotificationContent from './NotificationContent.svelte'
   import { removeNotification } from '../../store/actions'
-
   import type { Notification } from '../../types'
   import closeIcon from '../../icons/close-circle'
-
-  import { chainStyles, networkToChainId } from '../../utils'
   import { onDestroy, onMount } from 'svelte'
   import { configuration } from '../../configuration'
+  import { removeTransaction, transactions$, wallets$ } from '../../streams'
+  import type { Network } from 'bnc-sdk'
+  import { BigNumber } from 'ethers'
+  import { state } from '../../store'
 
-  const { device } = configuration
+  import {
+    chainStyles,
+    gweiToWeiHex,
+    networkToChainId,
+    toHexString
+  } from '../../utils'
+
+  const { device, gas, apiKey } = configuration
 
   export let notification: Notification
   export let updateParentOnRemove: () => void
 
   let timeoutId: NodeJS.Timeout
+  let hovered = false
+
+  // get transaction based on notification id
+  const transaction = transactions$
+    .getValue()
+    .find(({ hash }) => hash === notification.id)
+
+  // get wallet that sent transaction
+  const wallet =
+    transaction &&
+    $wallets$.find(
+      ({ accounts }) =>
+        !!accounts.find(
+          ({ address }) =>
+            address.toLowerCase() === transaction.from.toLowerCase()
+        )
+    )
 
   onMount(() => {
     if (notification.autoDismiss) {
       timeoutId = setTimeout(() => {
         removeNotification(notification.id)
+        removeTransaction(notification.id)
       }, notification.autoDismiss)
     }
   })
 
   onDestroy(() => {
     clearTimeout(timeoutId)
   })
+
+  function actionableEventCode(eventCode: Notification['eventCode']): boolean {
```

Comment:
@aaronbarnardsound Might be worth moving these functions into the notify.ts file or creating a new ts file for them like we did we preflight-notifications.ts...maybe even put all of them in a folder together to keep it clean...?

Reply:
Yeah agreed. Moved all of this logic out into a new `replacement.ts` file.